Kickstarter-iOS/Views/Controllers/SelectCurrencyViewController.swift

```diff
@@ -0,0 +1,79 @@
+import Library
+import Prelude
+import Prelude_UIKit
+import UIKit
+
+final class SelectCurrencyViewController: UIViewController, MessageBannerViewControllerPresenting {
+  private let viewModel: SelectCurrencyViewModelType = SelectCurrencyViewModel()
+
+  internal var messageBannerViewController: MessageBannerViewController?
+  private var saveButtonView: LoadingBarButtonItemView!
+
+  internal static func instantiate() -> SelectCurrencyViewController {
+    return SelectCurrencyViewController(nibName: nil, bundle: nil)
+  }
+
+  public func configure(with currency: Currency) {
+    self.viewModel.inputs.configure(with: currency)
+  }
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+
+    self.view.addSubview(self.tableView)
+    self.tableView.constrainEdges(to: self.view)
+
+    self.viewModel.inputs.viewDidLoad()
+  }
+
+  override func bindStyles() {
+    super.bindStyles()
+
+    _ = self.tableView
+      |> settingsTableViewStyle
+      |> \.separatorStyle .~ .singleLine
+  }
+
+  // MARK: Subviews
+
+  private lazy var tableView: UITableView = {
+    let tableView = UITableView(frame: .zero, style: .plain)
+
+    tableView.translatesAutoresizingMaskIntoConstraints = false
+    tableView.register(UITableViewCell.self, forCellReuseIdentifier: "cell")
+    tableView.tableFooterView = UIView(frame: .zero)
+    tableView.dataSource = self
+    tableView.delegate = self
+
+    return tableView
+  }()
+}
+
+extension SelectCurrencyViewController: UITableViewDataSource {
+  func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+    return Currency.allCases.count
+  }
+
+  func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+    let cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath)
+
+    let currency = Currency.allCases[indexPath.row]
+
+    cell.textLabel?.text = currency.descriptionText
+    cell.accessoryType = self.viewModel.outputs.isSelectedCurrency(currency) ? .checkmark : .none
+
+    return cell
+  }
+}
+
+extension SelectCurrencyViewController: UITableViewDelegate {
+  func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+    let currency = Currency.allCases[indexPath.row]
+
+    self.viewModel.inputs.didSelect(currency)
+
+    tableView.deselectRow(at: indexPath, animated: true)
+    tableView.visibleCells.forEach { $0.accessoryType = .none }
+    tableView.cellForRow(at: indexPath)?.accessoryType = .checkmark
```

Comment:
Hmm is there any way to test that the visual treatment of this cell is actually changing if its not coupled to an `output` on the view model? 🤔 This treatment feels like business logic to me, which would ideally live in the view model.

Reply:
One way to handle this would be to create an `output` for `selectedCurrency`:

```
self.viewModel.outputs.selectedCurrency
    .observeForUI()
    .observeValues { currency in
       // clear all accessory values && set the accessory on the correct cell here
    }
```

Then you can write a test with confidence that if the `selectedCurrency` output fires, then your cells are getting visually updated correctly.

Reply:
Going to run with @dusi's [suggestion](https://github.com/kickstarter/ios-oss/pull/569#discussion_r253633306), should be a good middle-ground.

Reply:
Due to the complexity of implementing this and fixing the tests later, I am going to do this refactor on #572 👍